tokio/src/stream/streamext/mod.rs

```diff
@@ -0,0 +1,190 @@
+use core::future::Future;
+use crate::stream::Stream;
+
+mod collect;
+mod filter;
+mod filter_map;
+mod map;
+mod next;
+
+#[allow(unreachable_pub)]
+pub use {
+    collect::Collect,
+    filter::Filter,
+    filter_map::FilterMap,
+    map::Map,
+    next::Next,
+};
+
+/// An extension trait for `Stream`s that provides a variety of convenient
+/// combinator functions.
+pub trait StreamExt: Stream {
```

Comment:
I'm also curious if it makes sense to include `try_next` in this ext trait, I find it to be super useful and is basically the only fn i use on `TryStreamExt` but I think we could include it here easily.

Reply:
I’m not sure if it’s really that useful since we have `Option::transpose`.

Reply:
I don't want to have to call transpose when I could just call `try_next`.

Reply:
We can discuss this in a follow up PR 👍